test/unit/gateways/secure_trading_test.rb

```diff
@@ -0,0 +1,321 @@
+require 'test_helper'
+
+class SecureTradingTest < Test::Unit::TestCase
+  def setup
+    @gateway = SecureTradingGateway.new(
+      user_id: 'user@test.com',
+      api_key: 'api_key',
+      site_id: 'site_id')
+
+    @credit_card = credit_card
+    @declined_card = credit_card('4000000000000002')
+    @amount = 100
+
+    @options = {
+      order_id: '1',
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_purchase
+    @gateway.expects(:ssl_post).returns(successful_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal '2-9-4019695', response.authorization
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    @gateway.expects(:ssl_post).returns(failed_purchase_response)
+
+    response = @gateway.purchase(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal Gateway::STANDARD_ERROR_CODE[:card_declined], response.error_code
+  end
+
+  def test_successful_authorize
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal '5-9-4895083', response.authorization
+    assert response.test?
+  end
+
+  def test_failed_authorize
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal '5-9-4895083', response.authorization
+    assert response.test?
+  end
+
+  def test_scrub
+    assert @gateway.supports_scrubbing?
+    assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
+  end
+
+  private
+
+  def pre_scrubbed
+    <<-PRE_SCRUBBED
+      opening connection to webservices.securetrading.net:443...
+      opened
+      starting SSL for webservices.securetrading.net:443...
+      SSL established
+      <- "POST /xml/ HTTP/1.1\r\nContent-Type: application/xml\r\nAuthorization: Basic ZXZlcmdpdmluZ0Bjb21taXR0ZWRnaXZpbmcudWsubmV0OktNKiVraiM3\r\nContent-Length: 989\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: Ruby\r\nConnection: close\r\nHost: webservices.securetrading.net\r\n\r\n"
```

Comment:
Always make sure that the credentials you use in the transcripts are not live credentials or sensitive ones right?